server/controllers/request.js

```diff
@@ -0,0 +1,99 @@
+const User = require("../models/User");
+const Profile = require("../models/Profile");
+const Request = require("../models/Request");
+
+exports.createRequest = async (req, res, next) => {
+    const { ownerId, sitterId } = req.body;
+
+    const ownerExists = await User.findOne({ ownerId });
+    if (!ownerExists) {
+        res.status(400);
+        throw new Error("Can't find any pet owner with this id");
+    }
+
+    const sitterExists = await User.findOne({ sitterId });
+    if (!sitterExists) {
+        res.status(400);
+        throw new Error("Can't find any petsitter with this id");
+    }
+
+    const ownerProfile = await Profile.findOne({ userId: ownerId });
+    const sitterProfile = await Profile.findOne({ userId: sitterId });
+    req.body.ownerName = ownerProfile.name;
+    req.body.sitterName = sitterProfile.name;
+    const request = await Request.create(req.body);
+    if (request) {
+        res.status(201).json(request);
+    } else {
+        res.status(400);
+        throw new Error("Invalid request data");
+    }
+};
+
+exports.getRequests = async (req, res, next) => {
+    const { userId } = req.body;
```

Comment:
It is not recommended to pass anything in the request body of a GET request. We can use the request object to obtain the logged in user's id 

Eg: `req.user.id`